### PR TITLE
Add support for IETF resumable upload draft

### PR DESCRIPTION
**Pull request by Acconut**
This patch provides a first, rough support for the new, upcoming resumable uploads draft. Details of its specification are currently available at https://datatracker.ietf.org/doc/draft-ietf-httpbis-resumable-upload/03/. With this PR we can gain first experience about the aspects and issues with implementing the draft.

This work is not yet ready for production. As details of the specification change, this PR will change as well.

### CODE REVIEW

File: lib/upload.js [hunk 1]
```diff
@@ -585,6 +595,9 @@ class BaseUpload {
       this._offset = 0
       promise = this._addChunkToRequest(req)
     } else {
+      if (this.options.protocol === PROTOCOL_IETF_DRAFT_01) {
+        req.setHeader('Upload-Incomplete', '?1')
```
Comment (smatsson) [inline lib/upload.js:599]: `Upload-Complete`,  `?0`

File: lib/upload.js [hunk 2]
```diff
@@ -810,6 +827,10 @@ class BaseUpload {
       if (value === null) {
         return this._sendRequest(req)
       }
+
+      if (this.options.protocol === PROTOCOL_IETF_DRAFT_01) {
+        req.setHeader('Upload-Incomplete', done ? '?0' : '?1')
```
Comment (smatsson) [inline lib/upload.js:832]: `Upload-Complete`, done ? `?1` : `?0`

File: lib/upload.js [hunk 3]
```diff
@@ -941,7 +962,11 @@ function inStatusCategory(status, category) {
 function openRequest(method, url, options) {
   const req = options.httpStack.createRequest(method, url)
 
-  req.setHeader('Tus-Resumable', '1.0.0')
+  if (options.protocol === PROTOCOL_IETF_DRAFT_01) {
+    req.setHeader('Upload-Draft-Interop-Version', '3')
```
Comment (smatsson) [inline lib/upload.js:966]: `req.setHeader('Upload-Draft-Interop-Version', '5')`

**Review by smatsson — COMMENTED** [review]
Reviewd changes to make it work with interop draft version 5. I did the same changes in the tus-js-client copy located in the tusdotnet test site

**Comment by Acconut** [comment]
Thank you very much, Stefan!

**Pull request merged**